src/main/java/org/wso2/okta/client/OktaOAuthClient.java

```diff
@@ -0,0 +1,1052 @@
+        appInfo.addParameter(OktaConstants.CLIENT_INITIATE_LOGIN_URI, initiateLoginUri);
+
+        return appInfo;
+    }
+
+    /**
+     * Revokes an access token.
+     *
+     * @param clientId     clientId of the oauth client
+     * @param clientSecret clientSecret of the oauth client
+     * @param accessToken  token being revoked
+     * @throws APIManagementException This is the custom exception class for API management.
+     */
+    private void revokeAccessToken(String clientId, String clientSecret, String accessToken)
+            throws APIManagementException {
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("Revoke access token from authorization Server. Access token: %s", accessToken));
+        }
+
+        CloseableHttpClient httpClient = HttpClientBuilder.create().build();
+        if (StringUtils.isEmpty(clientId)) {
+            handleException("Consumer Key can't be empty.");
+        }
+        if (StringUtils.isEmpty(clientSecret)) {
+            handleException("Consumer Secret can't be empty.");
+        }
+        if (StringUtils.isEmpty(accessToken)) {
+            handleException("Access Token can't be empty.");
+        }
+
+        try {
+            List<NameValuePair> nvps = new ArrayList<NameValuePair>();
+            nvps.add(new BasicNameValuePair(OktaConstants.TOKEN, accessToken));
+            nvps.add(new BasicNameValuePair(OktaConstants.TOKEN_TYPE_HINT, OktaConstants.ACCESS_TOKEN));
+
+            String oktaInstanceUrl = configuration.getParameter(OktaConstants.OKTA_INSTANCE_URL);
+            String authorizationServerId = configuration.getParameter(OktaConstants.OKTA_AUTHORIZATION_SERVER_ID);
+
+            HttpPost httpPost = new HttpPost(oktaInstanceUrl + OktaConstants.OAUTH2 + authorizationServerId +
+                    OktaConstants.REVOKE_ENDPOINT);
+            httpPost.setEntity(new UrlEncodedFormEntity(nvps));
+            String encodedCredentials = getEncodedCredentials(clientId, clientSecret);
+            httpPost.setHeader(OktaConstants.AUTHORIZATION, OktaConstants.AUTHENTICATION_BASIC + encodedCredentials);
+
+            if (log.isDebugEnabled()) {
+                log.debug("Invoking HTTP request to revoke access token.");
+            }
+            HttpResponse response = httpClient.execute(httpPost);
+            int statusCode = response.getStatusLine().getStatusCode();
+            if (statusCode == HttpStatus.SC_OK) {
+                if (log.isDebugEnabled()) {
+                    log.debug("OAuth accessToken has been successfully revoked.");
+                }
+            } else {
+                handleException(String.format("Problem occurred while revoking the accesstoken for Consumer Key %s",
+                        clientId));
+            }
+        } catch (UnsupportedEncodingException e) {
+            handleException(OktaConstants.ERROR_ENCODING_METHOD_NOT_SUPPORTED, e);
```

Comment:
These should NOT have constants, IMO. Why is it a constant?

Reply:
That same error message is used in multiple places. Sonar plugin suggested this.